scripts/prepdocslib/embeddings.py

```diff
@@ -139,18 +139,13 @@ def __init__(
         self.credential = credential
         self.cached_token: Optional[AccessToken] = None
 
-    async def create_embedding_arguments(self) -> dict[str, Any]:
-        return {
-            "model": self.open_ai_model_name,
-            "deployment_id": self.open_ai_deployment,
-            "api_type": self.get_api_type(),
-            "api_key": await self.wrap_credential(),
-            "api_version": "2023-05-15",
-            "api_base": f"https://{self.open_ai_service}.openai.azure.com",
-        }
-
-    def get_api_type(self) -> str:
-        return "azure_ad" if isinstance(self.credential, AsyncTokenCredential) else "azure"
+    async def create_client(self) -> AsyncOpenAI:
+        return AsyncAzureOpenAI(
+            azure_endpoint=f"https://{self.open_ai_service}.openai.azure.com",
+            azure_deployment=self.open_ai_deployment,
```

Comment:
Oh, interesting, you set deployment at the client level. I find that slightly unintuitive since I'm accustomed to deployment at function level but it does admittedly work well for this particular class, in terms of reducing the Azure/non-Azure differences, so this works. 👍🏼